### PR TITLE
[DT-962] Removes links from the access type column

### PR DESCRIPTION
**Pull request by s-rubenstein**
### Addresses
Removes links from access type. I decided to move to inlining the ternary because we didn't have a switch utility function - this ideally would be a switch statement.

Screenshot 2024-11-05 at 4 05 50 PM

### Summary

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/data_search/DatasetSearchTableConstants.tsx [hunk 1]
```diff
@@ -327,25 +327,16 @@ export const makeDatasetTableHeader = (datasets: DatasetTerm[], selected: number
...
-          label: `Access Type for dataset ${dataset.datasetId}: ${dataset.accessManagement}`
-        };
-      }
+      cellDataFn: (dataset: DatasetTerm) => ({
+        data: dataset.accessManagement === 'external' ?
+          'External to DUOS' :
+          dataset.accessManagement === 'open' ?
+            'Open Access' : dataset.dac?.dacName,
+        value: dataset.accessManagement,
+        id: `${dataset.datasetId}-participant-count`,
```
Comment (snf2ye) [inline src/components/data_search/DatasetSearchTableConstants.tsx:336]: Do you know why the id is named "participant-count" ?

File: src/components/data_search/DatasetSearchTableConstants.tsx [hunk 2]
```diff
@@ -327,25 +327,16 @@ export const makeDatasetTableHeader = (datasets: DatasetTerm[], selected: number
...
-          style: makeRowStyle(cellWidths.accessType),
-          label: `Access Type for dataset ${dataset.datasetId}: ${dataset.accessManagement}`
-        };
-      }
+      cellDataFn: (dataset: DatasetTerm) => ({
+        data: dataset.accessManagement === 'external' ?
+          'External to DUOS' :
+          dataset.accessManagement === 'open' ?
+            'Open Access' : dataset.dac?.dacName,
+        value: dataset.accessManagement,
```
Comment (snf2ye) [inline src/components/data_search/DatasetSearchTableConstants.tsx:335]: What is the difference between data and value?

**Review by snf2ye — APPROVED** [review]

File: src/components/data_search/DatasetSearchTableConstants.tsx (hunk 2 above)
Comment (fboulnois) [reply src/components/data_search/DatasetSearchTableConstants.tsx:335]: Data is what is shown to the user, value is the underlying content of the selection.

File: src/components/data_search/DatasetSearchTableConstants.tsx (hunk 1 above)
Comment (fboulnois) [reply src/components/data_search/DatasetSearchTableConstants.tsx:336]: Probably a copy-paste from a previous column. We should probably give this a unique name.

**Review by fboulnois — APPROVED** [review]
Looks good 👍

**Pull request merged**